### PR TITLE
Update documentation of Batch data plane swagger.

### PR DESCRIPTION
**Pull request by xingwu1**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by xingwu1** [comment]
@matthchr @darylmsft @dlepow Please review the doc change.

File: specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json
Comment (darylmsft) [inline specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json]: "and be": replace by "and will be" or "and"

File: specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json
Comment (darylmsft) [inline specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json]: "All tasks should be idempotent, it means..." replace by "All tasks should be idempotent.  This means tasks need to tolerate being... etc".  

I don't think the email on this text and/or placement has reached a conclusion yet.

File: specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json
Comment (hovsepm) [inline specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json]: Did you already ship client SDK with the current api version? Will they still work without this parameter being sent to server?

Comment (xingwu1) [reply specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json]: @hovsepm This is server response. Not request.

File: specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json
Comment (matthchr) [reply specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json]: We don't use PaaS/IaaS in public documentation.

Can you say this instead?
`On CloudServiceConfiguration pools, this user is logged in with the INTERACTIVE flag. On Windows VirtualMachineConfiguration pools, this user is logged in with the BATCH flag.`

File: specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json
Comment (matthchr) [reply specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json]: My attempt:
"Note that this value specifically controls the number of retries for the task executable due to nonzero exit code. The Batch service will try the task once, and may then retry up to this limit. For example, if the maximum retry count is 3, Batch tries the task up to 4 times (one initial try and 3 retries). If the maximum retry count is 0, the Batch service does not retry the task after the first attempt. If the maximum retry count is -1, the Batch service retries the task without limit. Resource files and application packages are only downloaded again if the task is retried on a new compute node. Batch will also retry tasks when a recovery operation is triggered on a compute node. Examples of recovery operations include (but are not limited to) when an unhealthy compute node is rebooted or a compute node disappeared due to host failure. Retries due to recovery operations are independent of and are not counted against the maxTaskRetryCount. Even if the maxTaskRetryCount is 0, an internal retry due to a recovery operation may occur. Because of this, all tasks should be idempotent. This means tasks need to tolerate being interrupted and restarted without causing any corruption or duplicate data. Best practices recommended for long running tasks is to use checkpointing." 

I think some/all of this text (specifically this bit: `Batch will also retry tasks when a recovery operation is triggered on a compute node. Examples of recovery operations include (but are not limited to) when an unhealthy compute node is rebooted or a compute node disappeared due to host failure. Retries due to recovery operations are independent of and are not counted against the maxTaskRetryCount. Even if the maxTaskRetryCount is 0, an internal retry due to a recovery operation may occur. Because of this, all tasks should be idempotent. This means tasks need to tolerate being interrupted and restarted without causing any corruption or duplicate data. Best practices recommended for long running tasks is to use checkpointing.` which is about internal retries should be moved to the root of the task object rather than buried down here on maxtaskRetryCount which is actually about _user_ retries not system/internal ones.

File: specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json
Comment (matthchr) [inline specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json]: Avoid this style of "reference" since it doesn't look nice in any generated language. I think we need to think of a better way to do this.

This comment applies down the line.

File: specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json
Comment (matthchr) [reply specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json]: Agree with Bevan.

**Review by matthchr — CHANGES_REQUESTED** [review]

File: specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json
Comment (xingwu1) [reply specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json]: We DO use PaaS/IaaS in the document, which exactly for cloudServiceConfiguration/virtualMachineConfig.

File: specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json
Comment (bgklein) [reply specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json]: Are we able to include new lines in the description documentation? The internal retries topic is really a separate concept than the user retries and it would be helpful to make a clear distinction when we branch off into it to prevent confusion.

I agree with Matt's point that idempotent tasks should be addressed  closer to the root as well because not everyone will care enough to read into retry policy. Generally when reading documentation people will start at the base and then delve deeper into the options as needed.

File: specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json
Comment (hovsepm) [reply specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json]: @xingwu1 why do you mark server response fields as required then? IMHO they should be marked as `"readOnly": true` to prevent client from modifying them. Required field means that those parameters should be present in the request and not response  - https://swagger.io/docs/specification/describing-parameters/

**Comment by hovsepm** [comment]
To all reviewers: please explicitly approve the PR then I'll be able to merge it.

File: specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json
Comment (xingwu1) [reply specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json]: readOnly and required field can't co-exist. However, without required field, all the int properties will convert to Nullable<int> which make user difficult to use. So right now, we will keep the required field right now. We will re-consider readonly field.

**Review by hovsepm — APPROVED** [review]

**Comment by hovsepm** [comment]
Waiting for all reviewers to sign-off.

**Comment by xingwu1** [comment]
@matthchr @dlepow Please review the PR and sign it off.

File: specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json
Comment (matthchr) [inline specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json]: Due to _a_ nonzero exit code (missing the "a")

File: specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json
Comment (matthchr) [inline specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json]: If the command line refers to file paths, it should use a relative path (relative to the task working directory), or use the Batch provided environment variables (https://docs.microsoft.com/en-us/azure/batch/batch-compute-node-environment-variables)

File: specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json
Comment (matthchr) [inline specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json]: At the end:
"The best practice for long running tasks is to use some form of checkpointing."

File: specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json
Comment (matthchr) [reply specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json]: This applies to all copies of this description, since there are a few

File: specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json
Comment (matthchr) [reply specification/batch/data-plane/Microsoft.Batch/stable/2018-03-01.6.1/BatchService.json]: Applies to all copies

**Review by matthchr — CHANGES_REQUESTED** [review]

**Review by matthchr — APPROVED** [review]

**Comment by xingwu1** [comment]
@hovsepm you can merge it

**Pull request merged**
